src/test/java/org/jbake/launcher/MainTest.java

```diff
@@ -0,0 +1,105 @@
+package org.jbake.launcher;
+
+import org.apache.commons.configuration.CompositeConfiguration;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import static org.mockito.Mockito.verify;
+
+@RunWith( MockitoJUnitRunner.class )
+public class MainTest {
+
+    private Main main;
+
+    @Mock private Baker mockBaker;
+    @Mock private JettyServer mockJetty;
+    @Mock private BakeWatcher mockWatcher;
+
+    @Before
+    public void setup() {
+        this.main = new Main(mockBaker, mockJetty, mockWatcher);
+    }
+
+    @Test
+    public void launchJetty() {
+        String[] args = {"-s"};
+        main.run(args);
+
+        verify(mockJetty).run("output","8820");
+    }
+
+    @Test
+    public void launchJettyWithCustomSourceDir() {
+        String[] args = {"src/jbake", "-s"};
+        main.run(args);
+
+        verify(mockJetty).run("output","8820");
+    }
+
+    // Documentation states these two commands will define the custom output, but the LaunchOptions file isn't setup for that.
+    // I have written this test to define the existing functionality of the code and not that defined in docs.
+    @Test
+    public void launchJettyWithCustomDestinationDir() {
+        String[] args = {"-s", "build/jbake"};
+        main.run(args);
+
+        verify(mockJetty).run("output","8820");
```

Comment:
This is the scenario I was referring to in my previous comment, in v2.5.0 the result of this execution `jbake -s build/jbake` would be that the contents of `build/jbake` would be served out not `output`. The underlying issue is the sharing of the source & destination values for different cmd line options.